spec/system/candidate_interface/references/reference_selection/candidate_reviews_references_spec.rb

```diff
@@ -0,0 +1,163 @@
+require 'rails_helper'
+
+RSpec.feature 'Review references' do
+  include CandidateHelper
+
+  before { FeatureFlag.activate(:reference_selection) }
+
+  scenario 'Candidate submits and reviews references' do
+    given_i_am_signed_in
+
+    when_i_have_no_references_and_try_to_visit_the_review_page
+    then_i_am_redirected_to_the_start_page
+
+    when_i_view_my_application
+    then_the_references_section_is_incomplete
+
+    when_i_have_added_references
+    then_the_references_section_is_still_incomplete
+
+    when_enough_references_have_been_provided_and_selected
+    and_i_mark_the_section_complete
+    then_the_references_section_is_complete
+    and_i_can_review_my_references_before_submission
+    and_i_can_delete_a_reference
+    and_i_can_delete_a_reference_request
+    and_i_can_cancel_a_sent_reference
+    and_referee_receives_email_about_reference_cancellation
+    and_i_can_return_to_the_application_page
+  end
+
+  def given_i_am_signed_in
+    create_and_sign_in_candidate
+  end
+
+  def when_i_have_no_references_and_try_to_visit_the_review_page
+    visit candidate_interface_references_review_path
+  end
+
+  def then_i_am_redirected_to_the_start_page
+    expect(page).to have_current_path candidate_interface_references_start_path
+  end
+
+  def when_i_view_my_application
+    visit candidate_interface_application_form_path
+  end
+
+  def then_the_references_section_is_incomplete
+    when_i_view_my_application
+    within '#select-your-references-badge-id' do
+      expect(page).to have_content 'Incomplete'
+    end
+  end
+
+  def then_the_references_section_is_still_incomplete
+    when_i_view_my_application
+    within '#select-your-references-badge-id' do
+      expect(page).to have_content 'Incomplete'
+    end
+  end
+
+  def when_i_have_added_references
+    application_form = current_candidate.current_application
+    @complete_reference = create(:reference, :feedback_provided, application_form: application_form)
+    @not_sent_reference = create(:reference, :not_requested_yet, application_form: application_form)
+    @requested_reference = create(:reference, :feedback_requested, application_form: application_form)
+    @refused_reference = create(:reference, :feedback_refused, application_form: application_form)
+    @cancelled_reference = create(:reference, :cancelled, application_form: application_form)
+    @bounced_reference = create(:reference, :cancelled, application_form: application_form)
+  end
+
+  def when_enough_references_have_been_provided_and_selected
```

Comment:
Works well! Only suggestion I have is perhaps a reword of this method to make it clearer that the references are auto selected (since there's only two) rather than it being an action the user takes? As this was my initial interpretation of it